File: ActivityName/app/src/main/java/com/codewithamit/activityname/MainActivity.java
package com.codewithamit.activityname;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ListView listView = (ListView)findViewById(R.id.listView);
      final   ArrayList<String> friends = new ArrayList<String>();

        friends.add("Amit");
        friends.add("Amit1");
        friends.add("Amit2");
        friends.add("Amit3");

        ArrayAdapter arrayAdapter = new ArrayAdapter(this, android.R.layout.simple_list_item_1,friends);

        listView.setAdapter(arrayAdapter);

        listView.setOnItemClickListener(new AdapterView.OnItemClickListener(){

            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {

                Intent intent =new Intent(getApplicationContext(),secpage.class);
                intent.putExtra("name",friends.get(i));

                startActivity(intent);
            }
        });

    }
}

File: Networkinginkotlin/app/src/main/java/com/example/networkinginkotlin/MainActivity.kt
package com.example.networkinginkotlin

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.appcompat.widget.SearchView
import androidx.recyclerview.widget.LinearLayoutManager
import com.google.gson.FieldNamingPolicy
import com.google.gson.GsonBuilder
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
import okhttp3.OkHttpClient
import okhttp3.Request
import org.json.JSONObject

class MainActivity : AppCompatActivity() {

    val originalList = arrayListOf<User>()
    val adapter = UserAdapter()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        adapter.onItemClick = {
            val intent = Intent(this, UserActivity::class.java)
            intent.putExtra("ID", it)
            startActivity(intent)
        }

        userRv.apply {
            layoutManager = LinearLayoutManager(this@MainActivity)
            adapter = this@MainActivity.adapter
        }

        searchView.isSubmitButtonEnabled=true
        searchView.setOnQueryTextListener(object :SearchView.OnQueryTextListener{
            override fun onQueryTextSubmit(query: String?): Boolean {
                query?.let{searchUsers(it)}
                return true
            }

            override fun onQueryTextChange(newText: String?): Boolean {
                newText?.let{searchUsers(it)}
                return true
            }
        })

        searchView.setOnCloseListener {
            adapter.swapdata(originalList)
            true
        }


        GlobalScope.launch(Dispatchers.Main) {
            val responce = withContext(Dispatchers.IO) { Client.api.getUser() }
            if (responce.isSuccessful) {
                responce.body()?.let {
                    originalList.addAll(it)
                    adapter.swapdata(it)
                }
            }
        }
    }
    fun searchUsers(query:String){

        GlobalScope.launch(Dispatchers.Main) {
            val responce = withContext(Dispatchers.IO){Client.api.searchUser(query)}
            if(responce.isSuccessful){
                responce.body()?.let {
                    it.items?.let { it1 -> adapter.swapdata(it1) }
                }
            }
        }
    }









//        val okHttpClient = OkHttpClient()
//
//        val request =  Request.Builder()
//            .url("https://api.github.com/users/amitmungare")
//            .build()
//
//        val gson = GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
//                .create()


//        GlobalScope.launch(Dispatchers.Main) {
//            val response = withContext(Dispatchers.IO){okHttpClient.newCall(request).execute().body?.string()}
//
//            val user = gson.fromJson<User>(response,User::class.java)
//
//            tv1.text = user.login
//            tv2.text = user.id.toString()
//            Picasso.get().load(user.avatarUrl).into(imageView)
//
//        }


}